spek-ide-plugin/interop-jvm/src/main/kotlin/org/spekframework/ide/ServiceMessageAdapter.kt

```diff
@@ -0,0 +1,113 @@
+
+        }
+        out("testFinished name='$name'")
+    }
+
+    override fun testIgnored(test: TestScopeImpl, reason: String?) {
+        val name = test.path.name.toServiceMessageSafeString()
+        out("testIgnored name='$name' ignoreComment='${reason ?: "no reason provided"}'")
+        out("testFinished name='$name'")
+    }
+
+    override fun groupExecutionStart(group: GroupScopeImpl) {
+        out("testSuiteStarted name='${group.path.name.toServiceMessageSafeString()}'")
+    }
+
+    override fun groupExecutionFinish(group: GroupScopeImpl, result: ExecutionResult) {
+        val name = group.path.name.toServiceMessageSafeString()
+        if (result is ExecutionResult.Failure) {
+            val exceptionDetails = getExceptionDetails(result)
+
+            // fake a child test
+            out("testStarted name='$name'")
+            out("testFailed name='$name' message='${exceptionDetails.first}' details='${exceptionDetails.second}'")
+            out("testFinished name='$name'")
+        }
+        out("testSuiteFinished name='$name'")
+    }
+
+    override fun groupIgnored(group: GroupScopeImpl, reason: String?) {
+        val name = group.path.name.toServiceMessageSafeString()
+        out("testIgnored name='$name' ignoreComment='${reason ?: "no reason provided"}'")
+        out("testFinished name='$name'")
+    }
+
+    override fun actionExecutionStart(action: ActionScopeImpl) {
+        out("testSuiteStarted name='${action.path.name.toServiceMessageSafeString()}'")
+    }
+
+    override fun actionExecutionFinish(action: ActionScopeImpl, result: ExecutionResult) {
+        val name = action.path.name.toServiceMessageSafeString()
+        if (result is ExecutionResult.Failure) {
+            val exceptionDetails = getExceptionDetails(result)
+
+            // fake a child test
+            out("testStarted name='$name'")
+            out("testFailed name='$name' message='${exceptionDetails.first}' details='${exceptionDetails.second}'")
+            out("testFinished name='$name'")
+        }
+        out("testSuiteFinished name='$name'")
+    }
+
+    override fun actionIgnored(action: ActionScopeImpl, reason: String?) {
+        val name = action.path.name.toServiceMessageSafeString()
+        out("testIgnored name='$name' ignoreComment='${reason ?: "no reason provided"}'")
+        out("testFinished name='$name'")
+    }
+
+    private fun getExceptionDetails(result: ExecutionResult.Failure): Pair<String?, String> {
+        val throwable = result.cause
+        val writer = CharArrayWriter()
+        throwable.printStackTrace(PrintWriter(writer))
+        val details = writer.toString()
+            .toServiceMessageSafeString()
+
+        val message = throwable.message?.toServiceMessageSafeString()
+
+        return message to details
+    }
+
+    private fun String.toServiceMessageSafeString(): String {

```

Comment:
Not sure I understand the purpose of this method.

Reply:
Encodes the string so it can be parsed safely by IntelliJ's service message parser.